### DISCUSSION TITLE
audio tag

### DISCUSSION THREAD

**Discussion by samyarmodabber** [mdx-js/mdx#2391] [Q&A] [answered]
Hello dear friends,
I want to add audio to mdx file but it does not show me and gives the following error:

The AudioContext was not allowed to start. It must be resumed (or created) after a user gesture on the page

Does anyone have a solution?

**Comment by wooorm** [comment]
This is likely unrelated to MDX, but to your framework, e.g., React, Next.js. Better to ask there.

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @samyarmodabber! 👋 
Sorry you ran into a spot of trouble.

The issue is you are trying to automatically play audio on a webpage when it loads.
This is **not allowed** by _browsers_:
* Chrome Policy https://developer.chrome.com/blog/autoplay/
* Firefox Policy https://developer.mozilla.org/en-US/docs/Web/Media/Autoplay_guide#autoplay_availability
* Safari Policy https://developer.apple.com/forums/thread/87881

Your options:
1. Don't autoplay audio, it is generally a bad user experience
2. If you need to play audio, have the user interact with the page in a way that satisfies the Chrome, Firefox, and Safari autoplay policies.

It is not a bug in MDX, React, a framework, or the browser. It is a protection browsers provide against bad user experience.
